pgspecial/iocommands.py

```diff
@@ -168,6 +183,8 @@ def execute_named_query(cur, pattern, **_):
         return [(None, None, None, message)]
 
     try:
+        if "$1" in query:
+            query, params = subst_favorite_query_args(query, params)
```

Comment:
Found a bug here. In case of missing substitution, you return `None` for the query. So you need a check for `if query is None`, and raise an error if that happens. Right now you'll get `can't execute an empty query` error, and that's not correct.

Reply:
add test on "query is None" raise Exception as bad arguments